wrappers/gin/gin_test.go

```diff
@@ -79,15 +92,66 @@ var _ = Describe("gin_wrapper", func() {
 					tracer.GlobalTracer.(*tracer.MockedEpsagonTracer).Labels["test"],
 				).To(Equal("ok"))
 			})
-			It("Creates a runner event for the handler invocation", func() {
-				testGinContext, _ := gin.CreateTestContext(httptest.NewRecorder())
+			It("Creates a runner and trigger events for handler invocation", func() {
 				mockedEngine.TestHandler = func(handler gin.HandlerFunc) {
 					handler(testGinContext)
 				}
 				wrapper.GET("/test", func(c *gin.Context) {
 					called = true
 				})
-				Expect(len(events)).To(Equal(1))
+				Expect(len(events)).To(Equal(2))
+			})
+			It("Adds correct trigger event", func() {
+				body := []byte("hello world")
+				testGinContext.Request = httptest.NewRequest(
+					"POST",
+					"https://www.help.com/test?hello=world&good=bye",
+					ioutil.NopCloser(bytes.NewReader(body)))
+				wrapper.Hostname = ""
+				wrapper.GET("/test", func(c *gin.Context) {
+					internalHandlerBody, err := ioutil.ReadAll(c.Request.Body)
+					if err != nil {
+						Expect(true).To(Equal(false))
+					}
+					Expect(internalHandlerBody).To(Equal(body))
+					called = true
+				})
+				Expect(len(events)).To(Equal(2))
+				var triggerEvent *protocol.Event
+				for _, event := range events {
+					if event.Origin == "trigger" {
+						triggerEvent = event
+					}
+				}
+				Expect(triggerEvent).NotTo(Equal(nil))
+				Expect(triggerEvent.Resource.Name).To(Equal("www.help.com"))
```

Comment:
can you also verify the resource type?

Reply:
if it is so important, even though I don't see value in verifying hardcoded values being passed around :-)

Reply:
I think checking the runner event resource type can be good here as a sanity test for the runner event

Reply:
This is a test for the trigger event, I think it is not wise to have one test case to test too many things. I have added the other verification that you asked for (Resource.Type) and others (Operation etc..)